packages/venia-concept/src/components/ProductQuantity/quantity.js

```diff
@@ -1,41 +1,86 @@
-import React, { Component } from 'react';
+import React from 'react';
+import { Text } from 'informed';
+import { Plus, Minus } from 'react-feather';
+import classify from 'src/classify';
 import { arrayOf, number, shape, string } from 'prop-types';
 
-import classify from '../../classify';
-import Select from '../Select';
-import mockData from './mockData';
 import defaultClasses from './quantity.css';
 
-class Quantity extends Component {
-    static propTypes = {
-        classes: shape({
-            root: string
-        }),
-        items: arrayOf(
-            shape({
-                value: number
-            })
-        )
+const Quantity = props => {
+    const { classes, initialValue, onValueChange, formApi, formState } = props;
+
+    const incrementQty = formApi => {
+        var currentQuantity = parseInt(formApi.getValue('quantity')) || 0;
+        const incrementedQty = currentQuantity + 1;
+        formApi.setValue('quantity', incrementedQty);
+        onValueChange(formApi.getValue('quantity'));
     };
 
-    static defaultProps = {
-        selectLabel: "product's quantity"
+    const decrementQty = formApi => {
+        var currentQuantity = parseInt(formApi.getValue('quantity')) || 0;
+        const decrementedQty = currentQuantity - 1;
+        formApi.setValue('quantity', decrementedQty);
+        onValueChange(formApi.getValue('quantity'));
     };
 
-    render() {
-        const { classes, selectLabel, ...restProps } = this.props;
+    const validateQuantity = value => {
+        var validateRegex = /^[1-9]\d*(\.\d+)?$/;
+        return value <= 0
```

Comment:
Can you rewrite this to not use a nested ternary? As written it is much harder to read.